code/game/objects/items/scanners.dm

```diff
@@ -80,21 +80,35 @@ REAGENT SCANNER
 
 /obj/item/healthanalyzer/attack(mob/living/carbon/M, mob/living/user)
 	. = ..()
+	analyze_vitals(M, user)
+
+/obj/item/healthanalyzer/attack_alternate(mob/living/carbon/M, mob/living/user)
+	. = ..()
+	analyze_vitals(M, user, TRUE)
+
+///Health scans a target. M is the thing being scanned, user is the person doing the scanning, show_patient will show the UI to the scanee when TRUE.
+/obj/item/healthanalyzer/proc/analyze_vitals(mob/living/carbon/M, mob/living/user, show_patient)
 	if(user.skills.getRating(SKILL_MEDICAL) < skill_threshold)
 		to_chat(user, span_warning("You start fumbling around with [src]..."))
 		if(!do_mob(user, M, max(SKILL_TASK_AVERAGE - (1 SECONDS * user.skills.getRating(SKILL_MEDICAL)), 0), BUSY_ICON_UNSKILLED))
 			return
 	playsound(src.loc, 'sound/items/healthanalyzer.ogg', 50)
-	if(CHECK_BITFIELD(M.species.species_flags, NO_SCAN))
-		to_chat(user, span_warning("Error: Cannot read vitals!"))
+	if(!iscarbon(M))
+		balloon_alert(user, "Cannot scan")
 		return
 	if(isxeno(M))
-		to_chat(user, span_warning("[src] can't make sense of this creature!"))
+		balloon_alert(user, "Unknown entity")
 		return
-	to_chat(user, span_notice("[user] has analyzed [M]'s vitals."))
+	if(CHECK_BITFIELD(M.species.species_flags, NO_SCAN))
+		balloon_alert(user, "Not Organic")
+		return
+	balloon_alert_to_viewers("Analyzed Vitals")
 	patient = M
 	current_user = user
-	ui_interact(user)
+	if(show_patient == TRUE)
+		ui_interact(M)
```

Comment:
Nah they dont get the updating med info, since there are already multiple checks being done to STOP_PROCESSING.
![image](https://user-images.githubusercontent.com/66163761/226597073-26489025-e5c4-4625-bce2-623b7567d904.png)
What this means is that the person you show their health scan will get a picture of what their scan was at the time of the scan. In order for them to get an updated scan, the person has to be scanned again after they've closed the UI.